php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -1464,43 +1463,39 @@ abstract class NDB_BVL_Instrument extends NDB_Page
     /**
      * Gets the sessionID for the timepoint to which this instrument pertains
      *
-     * @return int The sessionID if found, or -1 if not found
+     * @return SessionID The sessionID
+     * @throws NotFound
      */
-    function getSessionID(): int
+    function getSessionID(): SessionID
     {
-        $query      = "SELECT SessionID FROM flag WHERE CommentID = :CID";
-        $fieldValue = \Database::singleton()->pselectOne(
+        $query  = "SELECT SessionID FROM flag WHERE CommentID = :CID";
+        $result = \NDB_Factory::singleton()->database()->pselectOne(
             $query,
             [
                 'CID' => $this->getCommentID(),
             ]
         );
-
-        if (!empty($fieldValue)) {
-            return $fieldValue;
+        if (empty($result)) {
```

Comment:
The semantics of this function shouldn't be changed to throw an exception. This isn't backwards compatible and is unrelated to using the SessionID class.

Reply:
It has to be changed in order to use SessionID as a return type. Otherwise this function can return `?SessionID` but then a class like TimePoint that is created based on a SessionID will have to throw this Exception anyway.

It might not be backward compatible but this is going into a major release. Throwing an exception with a debug message seems greatly preferable to weakening the type-checking and returning `-1` without any context

Reply:
@driusan

Reply:
@driusan. Reverted the unrelated changes.

Reply:
I don't think throwing exception is a big deal for upgrading IBIS and I agree with John.

Reply:
Exception is not allowed here.